### PR TITLE
Added the merchant id to the query options

### PR DESCRIPTION
**Pull request by d1b1**
Added the merchant ID to allow and additional query. 

 client.itemSearch({
    keywords: req.query.keywords,
    ItemPage: 5,
    MerchantId: 'Amazon',
    searchIndex: 'HealthPersonalCare',
    responseGroup: 'ItemAttributes,Offers,Images'
  }, function(err, results) {
    if (err) {
      res.send(err.Error)
    } else {
      var data = results;

      if (showRaw == 'no') {
        var data = _.map(results, function(v) {
          return pluckItem(v);
        });
      }

      res.send(data);
    }
  });

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ var generateQueryString = function(query, method, credentials) {
   var searchIndex = query.searchIndex || 'All';
   var responseGroup = (query.responseGroup || 'ItemAttributes').replace(/,/g, '%2C');
   var condition = query.condition || 'All';
+  var merchant = query.MerchantId || 'All';
```
Comment (t3chnoboy) [inline lib/utils.js]: The attribute name should start with a lowercase letter.
Also, why the default value is 'All' ? I think we should skip it if no value is supplied.

**Comment by t3chnoboy** [comment]
Very nice! Thank you!
Can you please fix the issue with the default value (see the comment) and update README?

**Comment by FdezRomero** [comment]
@d1b1 It would be awesome if you can give this PR the finishing touches (change one letter in the attribute name and add a line to the Readme). It's almost there!

**Comment by d1b1** [comment]
Ok it tonight



Stephan Smith617-953-7847

**Comment by masterT** [comment]
This PR is no longer necessary, since the version 0.3.0 you can use all the param available for the methods, which includes the **MerchantId**.

**Comment by FdezRomero** [comment]
@masterT I just tried an ItemLookup with both `merchantId: 'Amazon'`and `MerchantId: 'Amazon'` for ASIN B00WT78QZS and it didn't work, it showed an offer with a merchant name that wasn't Amazon.

I checked the code and couldn't find why it's not working. Any help? Should I open an issue?

**Comment by masterT** [comment]
Can you show me your query and the value of the variable params at line 97 in the file `lib/utils.js` (just log it in the console).

**Comment by FdezRomero** [comment]
@masterT I didn't have line 97 in my `lib/utils.js` and noticed that `packages.json` had version 0.2.4. For some reason neither `npm update --save` nor `npm install amazon-product-api --save` were upgrading to version 0.3.1. I had to do `npm install amazon-product-api@0.3.1 --save` to make it work.

Anyway, it's working now so thanks a lot! I agree this PR is no longer needed.

**Comment by masterT** [comment]
Awesome :+1:!

**Comment by lukezona** [comment]
Sorry to bother you but I noticed that the MerchantId parameter is not available anymore.

How can I add it again? Am I just missing something?

**Pull request closed without merging**